game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java

```diff
@@ -262,52 +255,20 @@ public AggregateResults calculate(
                           bombarding,
                           territoryEffects,
                           retreatWhenOnlyAirLeft,
-                          currentWorkedRunCount));
-          list.add(workerResult);
+                          currentWorkedRunCount)));
         }
         runCount -= workerRunCount;
       }
       // Wait for all worker futures to complete and combine results
       final AggregateResults results = new AggregateResults(totalRunCount);
-      final Set<InterruptedException> interruptExceptions = new HashSet<>();
-      final Map<String, Set<ExecutionException>> executionExceptions = new HashMap<>();
       for (final Future<AggregateResults> future : list) {
         try {
           final AggregateResults result = future.get();
           results.addResults(result.getResults());
         } catch (final InterruptedException e) {
           Thread.currentThread().interrupt();
-          interruptExceptions.add(e);
         } catch (final ExecutionException e) {
-          final String cause = e.getCause().getLocalizedMessage();
-          Set<ExecutionException> exceptions = executionExceptions.get(cause);
-          if (exceptions == null) {
-            exceptions = new HashSet<>();
-          }
-          exceptions.add(e);
-          executionExceptions.put(cause, exceptions);
-        }
-      }
-      // we don't want to scare the user with 8+ errors all for the same thing
-      if (!interruptExceptions.isEmpty()) {
-        log.log(
-            Level.SEVERE,
-            interruptExceptions.size() + " Battle results workers interrupted",
-            interruptExceptions.iterator().next());
-      }
-      if (!executionExceptions.isEmpty()) {
-        Exception e = null;
-        for (final Set<ExecutionException> entry : executionExceptions.values()) {
-          if (!entry.isEmpty()) {
-            e = entry.iterator().next();
-            log.log(
-                Level.SEVERE,
-                entry.size() + " Battle results workers aborted by exception",
-                e.getCause());
-          }
-        }
-        if (e != null) {
-          throw new IllegalStateException(e.getCause());
+          throw new IllegalStateException("Battle results workers aborted by exception", e);
```

Comment:
nit, we probably should avoid the word "exception" from being displayed to users. It makes enough sense, but it's not the typical parlance of a player, they're not developers. It is a odd that the audience for most exceptions/error message is not developers but players.